src/states/node.rs

```diff
@@ -1272,7 +1260,7 @@ impl Node {
             return Ok(());
         }
 
-        let public_ids = match self.peer_mgr.routing_table().close_nodes(&dst_name, GROUP_SIZE) {
+        let public_ids = match self.peer_mgr.routing_table().close_names(&dst_name) {
             Some(close_group) => self.peer_mgr.get_pub_ids(&close_group),
```

Comment:
Possibly `get_pub_ids` should even be generic in its return type, to avoid the `into_iter().collect()` below.